tests/functional_tests/iterator_tests/test_builtins.py

```diff
@@ -108,6 +110,168 @@ def test_arithmetic_and_logical_builtins(backend, builtin, inputs, expected):
         assert np.allclose(np.asarray(out), expected)
 
 
+@pytest.mark.parametrize(
+    "math_fun, ref_impl, inputs",
+    [
+        # FIXME(ben): what about pow?
+        # FIXME(ben): dataset is missing invalid ranges (mostly nan outputs)
+        # FIXME(ben): we're not properly testing different datatypes
```

Comment:
I think this should be split into a follow up task. Other tests don't test with different datatypes either. Could probably go into a task as part of a bigger testing improvements effort.